PowerPointLabs/Test/UnitTest/SyncLab/SyncLabArtisticEffectTest.cs

```diff
@@ -4,48 +4,47 @@
-    /// PPT2013 exhibits strange behavior on ArtisticEffect.
-    /// ArtisticEffect is sometimes made permenant on pictures after saving the file.
-    /// i.e, image's artistic effect cannot be removed/undone
+    /// Be wary when using PPT2013 to edit the source powerpoint.
+    /// ArtisticEffect is sometimes made permenant on PPT2013
+    /// i.e, image's artistic effect cannot be removed/undone after the image is copied
     /// </summary>
     [TestClass]
-    public class SyncLabVisualEffectsTest : BaseSyncLabTest
+    public class SyncLabArtisticEffectTest : BaseSyncLabTest
     {
-        private const int OriginalReplaceEffectSlideNo = 4;
-        private const int OriginalSetEffectSlideNo = 6;
-        private const int DesiredReplaceEffectSlideNo = 7;
+        private const int OriginalSetEffectSlideNo = 4;
         private const int DesiredSetEffectSlideNo = 5;
+        private const int OriginalReplaceEffectSlideNo = 6;
+        private const int DesiredReplaceEffectSlideNo = 7;
         
         private const string SourceShape = "Picture 2";
         private const string SetArtisticEffect = "Picture 9";
         private const string ReplacedArtisticEffect = "Picture 10";
         
         protected override string GetTestingSlideName()
         {
-            return "SyncLab\\SyncLab_VisualEffects.pptx";
+            return "SyncLab\\SyncLab_ArtisticEffect.pptx";
         }
-        
+
         [TestMethod]
         [TestCategory("UT")]
-        public void TestReplaceArtisticEffect()
+        public void TestSetArtisticEffect()
         {
-            SyncArtisticEffect(ReplacedArtisticEffect, OriginalReplaceEffectSlideNo, DesiredReplaceEffectSlideNo);
+            SyncArtisticEffect(SetArtisticEffect, OriginalSetEffectSlideNo, DesiredSetEffectSlideNo);
         }
-
+        
         [TestMethod]
         [TestCategory("UT")]
-        public void TestSetArtisticEffect()
+        public void TestReplaceArtisticEffect()
         {
-            SyncArtisticEffect(SetArtisticEffect, OriginalSetEffectSlideNo, DesiredSetEffectSlideNo);
+            SyncArtisticEffect(ReplacedArtisticEffect, OriginalReplaceEffectSlideNo, DesiredReplaceEffectSlideNo);
         }
 
-        private void SyncArtisticEffect(string shapeToPaste, int sourceSlideNumber, int expectedSlideNo)
+        private void SyncArtisticEffect(string shapeToSync, int sourceSlideNumber, int expectedSlideNo)
         {
             Microsoft.Office.Interop.PowerPoint.Shape formatShape = GetShape(sourceSlideNumber, SourceShape);
 
-            Microsoft.Office.Interop.PowerPoint.Shape newShape = GetShape(sourceSlideNumber, shapeToPaste);
-            new PictureEffectsFormat().SyncFormat(formatShape, newShape);
+            Microsoft.Office.Interop.PowerPoint.Shape newShape = GetShape(sourceSlideNumber, shapeToSync);
+            new ArtisticEffectFormat().SyncFormat(formatShape, newShape);
 
             CompareSlides(sourceSlideNumber, expectedSlideNo);
```

Comment:
Is this actually enough to compare if the sync actually succeeded? If I remember correctly, this comparison is quite lenient and sometimes we have to make the shape differences very obvious to test it this way. I would suggest we try to test if the attributes directly (if possible) like using IsSameShape, although we would have to add those attributes to be tested too.

Reply:
I think it's sufficient as the artistic effects we are using for the test greatly changes the look of each shape. I'll look at it once more and check back just to be sure.

Reply:
Test for Replacing Effects:
![image](https://user-images.githubusercontent.com/11351531/38537015-cfb834ae-3cbe-11e8-9530-c24f2e25ba48.png)

The shape in the top left has it's effects changed after syncing.


Test for Setting Effects:
![image](https://user-images.githubusercontent.com/11351531/38537074-244643da-3cbf-11e8-8c7b-19a1179ef99b.png)

We set a checkered effect here. It also causes the overall lighting of the shape to change as well. 

I believe the change in images is large enough to be sufficient when we compare the slides.

Reply:
Also note that ArtisticEffect is only applicable to picture type shapes. I'll be adding this in as a comment as well.